### ISSUE TITLE
Add 'CKEditor Anchor Link - For Drupal 8' module to GovCMS8

### ISSUE THREAD

**Issue by ConradBem** [govCMS/govcms8-installer#393]
** What value does this module/package add to GovCMS?
This module makes it easy for a user to add and modify anchor links without knowledge of HTML. Presently, the only way to add anchor links is to manually add them in the source view.

** Please provide a brief outline of what this module does.
This module improves the WYSIWYG menu by adding a button to define an anchor on selected content. It also lets you point links to the anchors using the link button.

https://www.drupal.org/project/anchor_link

** Who does this module benefit:
[X] end users
[X] content editors
[X] site builders
[X] themers
[ ] developers
 
** How could you provide/replicate the functionality of this module using alternative methods, eg in your theme?
I would go into the source view of a page and add the anchors myself using HTML.

** If this module styles or alters HTML or JavaScript output, can the functionality be provided via the theme? What alternatives have you considered.
I don't believe a theme can replicate what this module does.

** What is the maintenance and support status of the module. Describe the issue queue activity.
The module's builder considers it feature complete, so the last stable release was in 2017.

There are 19 open issues and 12 open bugs.

** What permissions are needed to utilise the module (and are any new permissions provided by the module)?
I'm unsure what permissions are required, but the following modules are required to make this one work:
- CKEditor
- CKEditor FakeObjects module.
- CKEditor Anchor link plugin library. locate it in /libraries/ckeditor/plugins/link
- CKEditor Facke Objects plugin library. locate it in /libraries/fakeobjects, which managed by the CKEditor fakeobjects module.

There's also the following suggestions for version use:
- Drupal 8.5 >= use 8.x-2.0-alpha1
- Drupal 8.4 >= use 8.x-1.6
- Drupal 8.3 <= use 8.x-1.5

** Does the module modify the database structure and/or store additional metadata on nodes or other entities? If so, why? What are the risks for future updates?
I don't know, unfortunately.

** Is the module designed to capture anonymous user data?
I don't believe so.

** Is the output of the module typically fully cacheable? Would the inclusion of this module potentially render pages uncacheable.
I don't know, unfortunately.

** What is your assessment of the quality of this module, the contribution history of the module's maintainers, and the uptake of the module within the Drupal community?
According to Drupal.org, 13291 websites are using the module.

** Additional information

**Comment by simesy** [comment]
My review: The code is very simple and does very little. Adds a js file to the wysiwyg and defines some buttons and such.
-- no impact on frontend and anon performance
-- no modification of data 
-- code looks clean
-- would uninstall cleanly if needed

It has a dependency on [fakeobjects](https://www.drupal.org/project/fakeobjects) module which helps ckeditor provide other functionality. The review of this module is the same however, it's very simple and just provides ckeditor functionality.

I think this one doesn't have any effective work arounds and lack of this functionality has presented a lot of issues for editors I've worked with.

**Comment by simesy** [comment]
Needs investigation to be sure there are not alternative modules and solutions.

**Comment by joshua-graham-adelphi** [comment]
This issue seems to be a duplicate of Issue #315

**Comment by ConradBem** [comment]
Indeed, the module I've suggested is exactly the same one as in #315. Is there a way to merge this thread with that one?

**Comment by Belhendo** [comment]
Thank you for the request @ConradBem - we will review internally and let you know.

**Comment by ConradBem** [comment]
Hello @Belhendo, just checking in on this module request. Would you have an approximate idea of when the internal review will be complete?

**Comment by Belhendo** [comment]
Hi @ConradBem 

We have the linkit module (https://www.drupal.org/project/linkit) in the distro as a way to add links (external/internal) from the wysiwyg.  Are you able to articulate why you want this specific module  please.

**Comment by ConradBem** [comment]
Hello @Belhendo,

The Linkit module doesn't introduce any anchor linking functionality, unless I'm mistaken. Anchor linking is useful functionality, so I'm proposing the Anchor Link module as a solution.

**Comment by simesy** [comment]
> The Linkit module doesn't introduce any anchor linking functionality

This is correct

**Comment by Belhendo** [comment]
Thanks @simesy and @ConradBem.

We'll have a look at the impact and risks around implementing CKEditor Anchor Link with Linkit already in the distro. To help expidite this process, any insights about the interaction between these modules that you might have would be useful. Also any insights regarding D9 compatibility would be appreciated.
Thank you!

**Comment by simesy** [comment]
While it serves a different purpose to Linkit, I agree there are some question marks around compatibility with linkit. https://www.drupal.org/project/anchor_link/issues/2958532

Having looked more closely at this module it does fill a huge hole though.
